src/EditorFeatures/Core/CommandHandlers/InteractivePasteCommandHandler.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+extern alias InteractiveWindow;
+
+using System;
+using System.ComponentModel.Composition;
+using System.Runtime.CompilerServices;
+using System.Windows;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Text.Operations;
+using Microsoft.VisualStudio.Utilities;
+using Microsoft.CodeAnalysis.Editor.Commands;
+
+namespace Microsoft.CodeAnalysis.Editor.CommandHandlers
+{
+    [ExportCommandHandler(PredefinedCommandHandlerNames.InteractivePaste, ContentTypeNames.RoslynContentType)]
+    [Order(After = PredefinedCommandHandlerNames.Rename)]
+    [Order(After = PredefinedCommandHandlerNames.FormatDocument)]
+    [Order(After = PredefinedCommandHandlerNames.Commit)]
+    [Order(Before = PredefinedCommandHandlerNames.Completion)]
+    internal sealed class InteractivePasteCommandHandler : ICommandHandler<PasteCommandArgs>
+    {
+        // Duplicated string, originally defined at `Microsoft.VisualStudio.InteractiveWindow.PredefinedInteractiveContentTypes`
+        private const string InteractiveContentTypeName = "Interactive Content";
+        // Duplicated string, originally defined at `Microsoft.VisualStudio.InteractiveWindow.InteractiveWindow`
+        private const string InteractiveClipboardFormat = "89344A36-9821-495A-8255-99A63969F87D";
+
+        private readonly IEditorOperationsFactoryService _editorOperationsFactoryService;
+        private readonly ITextUndoHistoryRegistry _textUndoHistoryRegistry;
+
+        // This is for unit test purpose only, do not explicitly set this field otherwise.
+        internal IRoslynClipboard RoslynClipBoard;
```

Comment:
Won't change it to using MEF now sicne it does't affect functionality. Need to check this in and run perf test ASAP so we can get the fix in stabilization.